src/adsbinarynode.jl

```diff
@@ -0,0 +1,402 @@
+# This file is part of the TaylorIntegration.jl package; MIT licensed
+
+using StaticArrays: SVector
+import AbstractTrees: HasNodeType, NodeType, ParentLinks, StoredParents, TreeIterator,
+                      children, nodetype, nodevalue, parent, printnode, PreOrderDFS
+import Base: IteratorEltype, HasEltype, in, eltype, split
+
+if !isdefined(Base, :isnothing)        # Julia 1.0 support
+    using AbstractTrees: isnothing
+end
+
+export ADSDomain
+
+"""
+    ADSDomain{N, T <: Real}
+
+`N` dimensional box used in Automatic Domain Splitting to represent the
+domain of `N` jet transport variables.
+
+# Fields
+
+- `lo::SVector{N, T}`: lower bounds.
+- `hi::SVector{N, T}`: upper bounds.
+"""
+struct ADSDomain{N, T <: Real}
+    lo::SVector{N, T}
+    hi::SVector{N, T}
+    # Inner constructor
+    function ADSDomain{N, T}(lo::SVector{N, T}, hi::SVector{N, T}) where {N, T <: Real}
+        @assert all(hi .> lo) "Each upper bound must be greater than the respective lower bound"
+        new{N, T}(lo, hi)
+    end
+end
+
+# Outer constructors
+ADSDomain(lo::SVector{N, T}, hi::SVector{N, T}) where {N, T <: Real} = ADSDomain{N, T}(lo, hi)
+
+ADSDomain() = ADSDomain{1, Float64}(SVector{1, Float64}(zero(Float64)), SVector{1, Float64}(one(Float64)))
+
+function ADSDomain(x::Tuple{T, T}...) where {T <: Real}
+    N = length(x)
+    return ADSDomain{N, T}(
+        SVector{N, T}(x[i][1] for i in eachindex(x)),
+        SVector{N, T}(x[i][2] for i in eachindex(x))
+    )
+end
+
+# Print method for ADSDomain
+# Example:
+# [-1.0, 1.0]×[-1.0, 1.0]
+function show(io::IO, s::ADSDomain{N, T}) where {N, T <: Real}
+    x = Vector{String}(undef, N)
+    for i in eachindex(x)
+        x[i] = string(SVector{2, T}(s.lo[i], s.hi[i]))
+    end
+    print(io, join(x, "×"))
+end
+
+# Return the diameter of each direction in s
+diams(s::ADSDomain{N, T}) where {N, T <: Real} = s.hi - s.lo
+# Return the lower bounds in s
+infimums(s::ADSDomain{N, T}) where {N, T <: Real} = s.lo
```

Comment:
Since `infimums` is essentially a getter for the `lo` field, I'd suggest to either rename this method to `lo`, or rename the field to `infima`. Either way, I think it's more common to see `infima` for the plural (same comment for `supremums`).

Reply:
I think the second option sounds cooler, so I'll rename this functions to `infima` and `suprema` respectively.